packages/run-protocol/src/vaultFactory/prioritizedVaults.js

```diff
@@ -106,15 +106,7 @@ export const makePrioritizedVaults = (reschedulePriceCheck = () => {}) => {
    */
   const removeVault = key => {
     const vault = vaults.removeByKey(key);
-    const debtToCollateral = currentDebtToCollateral(vault);
-    if (
-      !oracleQueryThreshold ||
-      ratioGTE(debtToCollateral, oracleQueryThreshold)
-    ) {
-      // don't call reschedulePriceCheck, but do reset the highest.
-      // This could be expensive if we delete individual entries in order. Will know once we have perf data.
```

Comment:
not sure if:
1. you're now confident this will be a perf issue
2. looking at perf of this will be tracked in some other way
3. you don't think it needs noting because if there are perf problems with this they'll be easily diagnosed
4. the comment was accidentally removed and will be replaced

Reply:
#4 when I integrated some set of fixes here.

I think the right implementation is to remember the key of the highest and only update then. I want that to be a separate checkin.